ll-input.js

```diff
@@ -11,235 +12,164 @@ import {typeography} from '@lingolive/styles/ll-typeography.js';
  *
  *
  * @customElement
  * @polymer
  * @demo demo/index.html Input States
  * @demo demo/form.html Form Layout
  */
-export class Input extends LitElement {
+export class Input extends FormElementText {
   /**
    * Declare the properties that will
    * trigger calls to `_render`
    */
   static get properties() {
     return {
-      label: String,
-      message: String,
-      errorMessage: String,
-      isInvalid: Boolean,
-      disbled: Boolean,
-      required: Boolean,
-
-      for: String,
-      type: String,
-      placeholder: String,
-      autocomplete: String,
-      name: String,
-      autofocus: Boolean,
-      pattern: String,
+      disbled: {type: Boolean},
+      type: {type: String},
+      pattern: {type: String},
+      showsPassword: {type: Boolean},
     };
   }
   /**
-   * The properties that effect the rendering of the
-   * generated input for listening in `_propertiesChanged`
+   * override the _tagname function in base class (FormElementText)
+   *
+   * @return {string}
    */
-  get _inputProperties() {
-    return [
-      'for',
-      'type',
-      'placeholder',
-      'required',
-      'autocomplete',
-      'autofocus',
-      'pattern',
-    ];
+  get _tagname() {
+    return 'input';
   }
   /**
    * Use the constructor to set defaults to your properties.
    */
   constructor() {
     super();
-    this.label = '';
-    this.message = '';
-    this.isInvalid = false;
+    this.disabled = false;
     this.type = 'text';
-
-    this._blurHandler = this._blurHandler.bind(this);
-    this._inputHandler = this._inputHandler.bind(this);
-    this._invalidHandler = this._invalidHandler.bind(this);
-  }
-  /**
-   * Cache the input on connect
-   */
-  connectedCallback() {
-    super.connectedCallback();
-    this.$input;
-  }
-  /**
-   * Find and cache the input element
-   *
-   * @return {element}
-   */
-  get $input() {
-    if (!this._$input) {
-      let input = this.querySelector('input');
-      if (!input) {
-        this._generatedInput = true;
-        this._renderInput();
-        input = this.querySelector('input');
-      } else {
-        input.removeEventListener('blur', this._blurHandler);
-        input.removeEventListener('input', this._inputHandler);
-        input.removeEventListener('invalid', this._invalidHandler);
-        input.addEventListener('blur', this._blurHandler);
-        input.addEventListener('input', this._inputHandler);
-        input.addEventListener('invalid', this._invalidHandler);
-      }
-      this._$input = input;
-    }
-    return this._$input;
+    this.showsPassword = false;
+    this._toggleShowsPassword = this._toggleShowsPassword.bind(this);
   }
   /**
-   * Updates the generated input element
-   */
-  _renderInput() {
-    render(this._input(), this, this.localName);
-  }
-  /**
-   * Build a list of classes for the message element from state.
+   * Allow `password` type inputs to toggle the visible of the input content
```

Comment:
```suggestion
   * Returns the display type that configures template behavior.
```

I think the comment as it is very specific to the implementation details of what it is now. If we wanted to update and add more behavior we'd have to change the comment.